client/public/js/components/EventDetail.js

```diff
@@ -37,19 +44,28 @@ var EventDetail = React.createClass({
       </div>
     );
   },
-
+  //created a prompt asking for waht the user wants to change the event to
   _editEvent: function(){
-    EventDetailActions.edit();
+    this.state.name = prompt('What should the Event be called?');

```

Comment:
I don't think we should be assigning values directly to `this.state` here on line 49 and line 59. Instead try this:

```
_editEvent: function(){
    var obj = {};
    obj.name = prompt('What should the Event be called?');
    obj.location = this.state.location;
    EventDetailActions.edit(obj);
  },
```

Do the similar thing with `_editLocation`